src/config.ts

```diff
@@ -60,32 +66,63 @@ export default class ConfigService {
       ? await loadViaHttp(this.#uri)
       : require(/* webpackIgnore: true */ this.#uri).default;
 
-    const [renderConfig, errors] = validateRenderConfig(config);
+    const [validConfig, errors] = validateConfig(config);
 
     if (errors !== undefined) {
       throw new Error(errors?.message);
     }
 
-    logger.info(`Render config valid. ${renderConfig.size} styles available.`);
+    logger.info(
+      `Render config valid. ${validConfig.renderConfig.size} styles available.`
+    );
 
-    this.#config = renderConfig;
+    this.#currentVersion = validConfig.version;
+    this.#renderConfig = validConfig.renderConfig;
   }
 
-  setConfig(style: string, config: RenderDescriptor) {
-    this.#config.set(style, config);
+  /**
+   * Set a specific configuration for rendering. Primarily used for testing.
+   *
+   * @internal
+   */
+  setRenderConfig(style: string, config: RenderDescriptor) {
+    this.#renderConfig.set(style, config);
+  }
+
+  /**
+   * Set the current config version. Primarily used for testing.
+   *
+   * @internal
+   */
+  setVersion(version: string) {
+    this.#currentVersion = version;
   }
 
   /**
    * Get a render style configuration given the style key
    */
   getConfig(style: string) {
-    return this.#config.get(style);
+    return this.#renderConfig.get(style);
   }
 
   /**
    * Get the list of available render style configurations
    */
-  renderStyles() {
-    return this.#config.keys();
+  get renderStyles() {
+    return this.#renderConfig.keys();
+  }
+
+  /**
+   * Checks that we have loaded a configuration file
+   */
+  get isLoaded() {
+    return this.#currentVersion !== null;
+  }
+
+  /**
+   * Get the current configuration version
+   */
+  get version() {
+    return this.#currentVersion ?? 'config-not-loaded';
```

Comment:
`'config-not-loaded'` here should probably be a constant with a `Symbol` to distinguish it from a regular string version. Very unlikely but it is still possible to set a version that actually is `'config-not-loaded'`.

Reply:
Oh good idea